cores/esp8266/Esp.cpp

```diff
@@ -107,11 +107,19 @@ void EspClass::wdtFeed(void)
 
 extern "C" void esp_yield();
 
-void EspClass::deepSleep(uint64_t time_us, WakeMode mode)
+bool EspClass::deepSleep(uint64_t time_us, WakeMode mode)
 {
+    if (time_us > deepSleepMax())
```

Comment:
Would it not be more useful (and help fix peoples code which is broken already) if, instead of possibly printing an error and `return false;` , you just printed a *warning* and adjusted the time down to `deepSleepMax()`? 

Existing code will not expect to ever get a return value and potentially go off into lala-land, since this was a `void`.

"Be conservative in what you do, be liberal in what you accept from others."

Reply:
@earlephilhower Point taken. I'll make some changes to the code.